pkg/operator/feature_gates.go

```diff
@@ -19,47 +19,110 @@ import (
 	"slices"
 	"strings"
 
-	k8sflag "k8s.io/component-base/cli/flag"
+	"github.com/prometheus/client_golang/prometheus"
 )
 
-// At the moment, the are no feature gates available.
-var defaultFeatureGates = map[string]bool{
-	"PrometheusAgentDaemonSet": false,
+const (
+	// PrometheusAgentDaemonSetFeature enables the DaemonSet mode for PrometheusAgent.
+	PrometheusAgentDaemonSetFeature FeatureGateName = "PrometheusAgentDaemonSet"
+)
+
+type FeatureGateName string
+
+type FeatureGates map[FeatureGateName]FeatureGate
+
+type FeatureGate struct {
+	description string
+	enabled     bool
 }
 
-// ValidateFeatureGates merges the feature gate default values with
+func (fg FeatureGates) Enabled(name FeatureGateName) bool {
```

Comment:
@ArthurSens @haanhvu I've added this so you can check feature gates being enabled using `cfg.FeatureGates.Enabled(operator.PrometheusAgentDaemonSetFeature)`.